src/libs/location/src/character.cpp

```diff
@@ -2828,21 +2931,48 @@ if(storm::iEquals(eventName, "Blade to belt"))
                 else if (storm::iEquals(eventName, CHARACTER_FIGHT_GUNHAND))
                 {
                     core.Send_Message(blade, "l", MSG_BLADE_GUNHAND);
+                    isFireState = true;
                 }
                 else if (storm::iEquals(eventName, CHARACTER_FIGHT_GUNFIRE))
                 {
+                    isFireState = false;
                     core.Send_Message(blade, "l", MSG_BLADE_GUNFIRE);
                     // PlaySound("pistol_shot");
                     isFired = true;
                     float kDist;
                     Character *chr = FindGunTarget(kDist, CheckShotOnlyEnemyTest());
                     entid_t enemy{};
+                    int32_t resHit = 0;
                     if (chr)
                     {
                         enemy = chr->GetId();
-                        chr->Hit(fgt_hit_fire);
+			resHit = 1;
+			VDATA *vd = core.Event("Check_ChrHitFire", "iilf", GetId(), enemy, static_cast<int32_t>(chr->isRecoilState), kDist);
+			if(vd)
+			{
+                            if(vd->Get(resHit))
+                            {
+                                if(resHit == 2)
+                                {
+                                    core.Event("Location_CharacterFire", "iifl", GetId(), enemy, kDist, 1);
```

Comment:
smells bad
`core.Event("Location_CharacterFire", "iifl", GetId(), enemy, kDist, ...);` is duplcated four times
Maybe add variable with last value and only change it?

Reply:
Sure. That was a little spaghetti because I didn't adapt my old dodge well for compatibility. Also "jedi bullet" was from here.